kafka-admin/src/main/java/io/strimzi/admin/kafka/admin/handlers/TopicDescriptionHandler.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.admin.kafka.admin.handlers;
+
+import io.strimzi.admin.kafka.admin.AdminClientProvider;
+import io.strimzi.admin.kafka.admin.model.Types;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.ext.web.handler.graphql.VertxDataFetcher;
+import io.vertx.kafka.admin.Config;
+import io.vertx.kafka.admin.ConfigEntry;
+import io.vertx.kafka.client.common.ConfigResource;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class TopicDescriptionHandler {
+
+    public static VertxDataFetcher topicDescriptionFetch(AdminClientProvider acp) {
+        VertxDataFetcher<Types.Topic> dataFetcher = new VertxDataFetcher<>((environment, prom) -> {
+            String topicToDescribe = environment.getArgument("name");
+            if (topicToDescribe == null || topicToDescribe.isEmpty()) {
+                prom.fail("Topic to describe has not been specified");
+            }
+            Promise<Map<String, io.vertx.kafka.admin.TopicDescription>> describeTopicsPromise = Promise.promise();
+            acp.describeTopics(Collections.singletonList(topicToDescribe), describeTopicsPromise);
+
+            ConfigResource resource = new ConfigResource(org.apache.kafka.common.config.ConfigResource.Type.TOPIC, topicToDescribe);
```

Comment:
this can be local to the lambda where it's used right? It's used just in the  `onComplete` where you know the topic name as well.